sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -161,51 +162,45 @@ namespace Azure { namespace IO {
   };
 
   /**
-   * @brief #Azure::IO::BodyStream providing its data from a file.
+   * @brief #Azure::IO::BodyStream used for reading data from a file.
    */
   class FileBodyStream : public BodyStream {
   private:
     // immutable
-#if defined(AZ_PLATFORM_POSIX)
-    int m_fd;
-#elif defined(AZ_PLATFORM_WINDOWS)
-    HANDLE m_hFile;
+    Azure::IO::Internal::FileHandleHolder m_fileStreamHolder;
```

Comment:
Kill this

Reply:
Will use try/catch to ensure the file gets closed if the ctor throws.